auth/utils/read-external-sources.ts

```diff
@@ -97,8 +96,31 @@ function filterPropertiesByServiceName(envObj: any, serviceName: string): any {
  * Pulls credentials from VCAP_SERVICES env property that IBM Cloud sets
  *
  */
-function getCredentialsFromCloud(serviceName: string): any {
-  const credentials = vcapServices.getCredentials(serviceName);
+function getVCAPCredentialsForService(name) {
+  if (process.env.VCAP_SERVICES) {
+    const services = JSON.parse(process.env.VCAP_SERVICES);
+    for (const serviceName of Object.keys(services)) {
```

Comment:
> for (const serviceName of Object.keys(services)) {

Isn't this the same thing as `for (const serviceName in services) {` below?

It might be more concise to combine the loops into something like:

```js
    for (const serviceName in services) {
      const service = services[serviceName];
      for (const instance of service) {
        if (instance.name === name) {
          return instance.credentials || {};
        }
      }
      if (serviceName === name) {
        if (service.length > 0) {
          return service[0].credentials || {};
        } else {
          return {}
        }
      }
    }
```

I just found myself as the reader a little confused why there were two different loops for the same thing. This snippet includes some other things to make the code a little more concise as well.

Reply:
Okay actually, there is an extremely unlikely edge case where my code would not work (if one of the top level names matches a specific "name" field under one of the other top level services. So the two loops are fine, but I would still use `for (const serviceName in services) {` for both for clarity.

Reply:
`for (const serviceName in services)` failed tslint with the error: `read-external-sources.ts[102, 5]: for (... in ...) statements must be filtered with an if statement`. But you're right in that the 2 for-loops does the same thing, but written differently. I'll changed the 2nd for-loop to match the first one to make it consistent.

Reply:
Huh, that's a weird linting rule, but fair enough. That sounds good